core/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```
@@ -294,10 +289,12 @@ public int compare(MyThreadInfo o1, MyThreadInfo o2) {
                 }
             }
             return sb.toString();
-        } finally {
-            if (enabledCpu) {
-                threadBean.setThreadCpuTimeEnabled(false);
-            }
+        } catch (Exception e) {
+            logger.warn("unable to capture hot_threads", e);
+            // Return as much as we were able to generate at least, plus the exception
+            sb.append("unable to capture hot_threads%n");
+            sb.append(ExceptionsHelper.stackTrace(e));
+            return sb.toString();
         }
```

Why do we need to hide exceptions here?